test/producer-consumer-spec.js

```diff
@@ -1,183 +1,199 @@
-      assert(response === true);
-      ++letters;
-    })
-    .then(function () {
-      return consumer.consume(fixtures.queues[0], function (_msg) {
-        assert(_msg === undefined);
-      });
-    })
-    .then(function (response) {
-      assert(response === true);
-      --letters;
-    })
-    .then(done);
+  after(() => {
+    return docker.stop();
   });
 
-  it('should be able to consume message sended by producer to queue [test-queue-0] (null message)', function (done) {
-    producer.produce(fixtures.queues[0], null)
-    .then(function (response) {
-      assert(response === true);
-      ++letters;
-    })
-    .then(function () {
-      return consumer.consume(fixtures.queues[0], function (_msg) {
-        assert(_msg === null);
-      });
-    })
-    .then(function (response) {
-      assert(response === true);
-      --letters;
-    })
-    .then(done);
-  });
+  describe('msg delevering', function() {
 
-  it('should not be able to consume message sended by producer to queue [test-queue-1]', function (done) {
-    producer.produce(fixtures.queues[1], { msg: uuid.v4() })
-    .then(function (response) {
-      assert(response === true);
-      ++letters;
-    })
-    .then(function () {
-      return consumer.consume(fixtures.queues[0], function (_msg) {
-        assert(typeof _msg === 'object');
-      });
-    })
-    .then(function (response) {
-      assert(response === true);
-      --letters;
-    })
-    .then(done);
-  });
+    it('should be able to consume message sended by producer to queue [test-queue-0]', function (done) {
+      producer.produce(fixtures.queues[0], { msg: uuid.v4() })
+      .then(function (response) {
+        assert(response === true);
+        ++letters;
+      })
+      .then(function () {
+        return consumer.consume(fixtures.queues[0], function (_msg) {
+          assert(typeof _msg === 'object');
+        });
+      })
+      .then(function (response) {
+        assert(response === true);
+        --letters;
+      })
+      .then(done);
+    });
 
-  it('should be able to consume message sended by producer to queue [test-queue-1], and the message of the previous test case', function (done) {
-    producer.produce(fixtures.queues[1], { msg: uuid.v4() })
-    .then(function (response) {
-      assert(response === true);
-      ++letters;
-    })
-    .then(function () {
-      return new Promise(function (resolve) {
-        consumer.consume(fixtures.queues[1], function (_msg) {
+    it('should be able to consume message sended by producer to queue [test-queue-0] (no message)', function (done) {
+      producer.produce(fixtures.queues[0])
+      .then(function (response) {
+        assert(response === true);
+        ++letters;
+      })
+      .then(function () {
+        return consumer.consume(fixtures.queues[0], function (_msg) {
+          assert(_msg === undefined);
+        });
+      })
+      .then(function (response) {
+        assert(response === true);
```

Comment:
OK I may be wrong about the flow here,  but if response here is false, it will throw assertion error and reject the promise and there is no  .catch section so this test still will fail but with a timeout and not , a more informative, error. If this is the case, consider adding .catch(done) or just return the promise chain to mocha and remove done altogether.